### PR TITLE
Apply Swan Lake Alpha4 migration changes

### PR DESCRIPTION
**Pull request by kapilraajP**
Resolves wso2-enterprise/choreo#3053

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: endpoint.bal [hunk 1]
```diff
@@ -54,10 +56,12 @@ public client class Client {
     # + queueName - Name of the queue to be created 
     # + attributes - Other attribute parameters 
     # + return - If success, URL of the created queue, else returns error
-    public remote function createQueue(string queueName, map<string> attributes)
-            returns @tainted string|OperationError{
-        string amzTarget = AMAZON_SQS_API_VERSION + "." + ACTION_CREATE_QUEUE;
-        string endpoint = "/";
+    @display {label: "Create queue"}
+    remote isolated function createQueue(@display {label: "Queue name"} string queueName, 
+                                @display {label: "Map of attributes"} map<string> attributes)
```
Comment (abeykoon) [inline endpoint.bal:61]: `Map of attributes` is not a meaningful name for a display name.

File: endpoint.bal [hunk 2]
```diff
@@ -90,19 +94,22 @@ public client class Client {
     # + queueResourcePath - Resource path to the queue from the host address. e.g.: /610968236798/myQueue.fifo
     # + attributes - Non-mandatory parameters for sending a message 
     # + return - If success, details of the sent message, else returns error
-    public remote function sendMessage(string messageBody, string queueResourcePath, map<string> attributes) 
-            returns @tainted OutboundMessage|OperationError {
-        string|error msgbody = encoding:encodeUriComponent(messageBody, UTF_8);
+    @display {label: "Send message in queue"}
+    remote isolated function sendMessage(@display {label: "Message body to send"} string messageBody, 
+                                @display {label: "Resource path to queue"} string queueResourcePath, 
+                                @display {label: "Map of attributes"} map<string> attributes)
```
Comment (abeykoon) [inline endpoint.bal:100]: Make attributes as optional. Are they mandatory?

Comment (abeykoon) [reply endpoint.bal:100]: Shall we model them as optional parameters? Then it is more convenient and user friendly from language perspective as well as UI perspective. 
Apply this for other operations as well where necessary.

File: endpoint.bal (hunk 1 above)
Comment (indikasampath2000) [reply endpoint.bal:61]: @kapilraajP, please follow the SQS API Reference doc and implement all operations according to that. The createQueue action is taking three inputs. But you have implemented only two. Also, you have missed the tag parameter as well. As @abeykoon mentioned, you have to set all not required parameters to optional.

File: endpoint.bal (hunk 2 above)
Comment (indikasampath2000) [reply endpoint.bal:100]: Please check my above comment. You have to follow up the API Reference doc and implement all parameters in the sendMessage operation.

File: endpoint.bal [hunk 3]
```diff
@@ -129,16 +136,18 @@ public client class Client {
     # + queueResourcePath - Resource path to the queue from the host address. e.g.: /610968236798/myQueue.fifo 
     # + attributes - Non-mandatory parameters for receiving a message
     # + return - If success, details of the received message, else returns error
-    public remote function receiveMessage(string queueResourcePath, map<string> attributes) 
-            returns @tainted InboundMessage[]|OperationError {
-        string amzTarget = AMAZON_SQS_API_VERSION + "." + ACTION_RECEIVE_MESSAGE;
+    @display {label: "Receive message in queue"}
+    remote isolated function receiveMessage(@display {label: "Resource path to queue"} string queueResourcePath,
```
Comment (indikasampath2000) [inline endpoint.bal:140]: Please check my above comment. You have to follow up the API Reference doc and implement all parameters in the receiveMessage operation.

File: endpoint.bal [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+            return error OperationError(OPERATION_ERROR_MSG, request);
+        }
+    }
+
+    private isolated function buildPayload(map<string> parameters) returns string {
+        string payload = EMPTY_STRING;
+        int parameterNumber = 1;
+        foreach var [key, value] in parameters.entries() {
+            if (parameterNumber > 1) {
+                payload = payload + AMBERSAND;
```
Comment (indikasampath2000) [inline endpoint.bal:293]: Wrong wording AMBERSAND must be AMPERSAND

**Comment by abeykoon** [comment]
Committing changes after an offline discussion with @indikasampath2000. Minor changes requested here should be performed after SLAplha4 release.

**Review by abeykoon — APPROVED** [review]

File: endpoint.bal (hunk 1 above)
Comment (kapilraajP) [reply endpoint.bal]: Fixed

File: endpoint.bal (hunk 2 above)
Comment (kapilraajP) [reply endpoint.bal]: Made those optional

File: endpoint.bal (hunk 3 above)
Comment (kapilraajP) [reply endpoint.bal:176]: Noted and Resolved

File: endpoint.bal (hunk 4 above)
Comment (kapilraajP) [reply endpoint.bal:293]: Corrected the typo

**Pull request merged**